.github/workflows/update-badge.yml

```
@@ -0,0 +1,59 @@
+# Note: this workflow requires the repository to give Write access to Github Workflows.
+# in Settings > Actions > General > Workflow permissions
+name: Update Badge
+
+on:
+  workflow_run:
+    workflows:
+      - Test Production (e2e)
+    types:
+      - completed
+    branches:
+      - main
+
+defaults:
+  run:
+    shell: bash
+
+concurrency:
+  group: ${{ github.workflow }}
+  cancel-in-progress: true
+
+jobs:
+  update-badge:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: pl-strflt/job-summary-url-action@v1
+        id: metadata
+        with:
+          workflow: test-prod-e2e.yml # ${{ github.event.workflow.path }}
+          run_id: ${{ github.event.workflow_run.id }}
+          run_attempt: ${{ github.event.workflow_run.run_attempt }}
+          job: aggregate
+      - uses: actions/checkout@v3
+      # https://github.com/orgs/community/discussions/26560
+      - run: |
+          git config user.email "41898282+github-actions[bot]@users.noreply.github.com"
+          git config user.name "github-actions[bot]"
+      - run: |
+          echo GITHUB_JOB_SUMMARY_URL=${GITHUB_JOB_SUMMARY_URL}
+          IN='[![Conformance Production Dashboard](https://github.com/ipfs/gateway-conformance/actions/workflows/test-prod-e2e.yml/badge.svg?branch=master)](.*)'
+          ESCAPED_IN=$(printf '%s\n' "$IN" | sed -e 's/[][\/!&]/\\&/g')
+
+          OUT="[![Conformance Production Dashboard](https://github.com/ipfs/gateway-conformance/actions/workflows/test-prod-e2e.yml/badge.svg?branch=master)](${GITHUB_JOB_SUMMARY_URL})"
+
+          sed -i "s;${ESCAPED_IN};${OUT};" README.md
+        env:
+          GITHUB_JOB_SUMMARY_URL: ${{ steps.metadata.outputs.job_summary_url }}
+          REPOSITORY: ${{ github.repository }}
+      - run: |
+          if [[ -n $(git diff --shortstat 2> /dev/null | tail -n1) ]]; then
+            echo "GIT_DIRTY=1" >> $GITHUB_ENV
+          else
+            echo "GIT_DIRTY=0" >> $GITHUB_ENV
+          fi
+      - if: env.GIT_DIRTY == '1'
```

I tend to favour output over env lately as it's harder to use it by mistake. Here, it doesn't make much difference to be honest. 

```suggestion
      - id: git
         run: |
          if [[ -n $(git diff --shortstat 2> /dev/null | tail -n1) ]]; then
            echo "dirty=1" >> $GITHUB_OUTPUT
          else
            echo "dirty=0" >> $GITHUB_OUTPUT
          fi
      - if: steps.git.outputs.dirty == '1'
```

---

That make sense, thanks for the recommendation!

FWIW I'm looking into generalizing the update-badge, I'll update it soon starting from your implem in libp2p/test-plans, wip version: https://github.com/singulargarden/gateway-conformance/blob/59ace38c4519d16a5cd11c79ecbf85dd09419374/.github/workflows/update-badge.yml#L35